.buildkite/build-ios.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash -eu
+
+echo '--- :node: Setup Node depenendencies'
+echo '--- :node: 1. Install nvm'
+brew install nvm
+
+echo '--- :node: 2. Load nvm in the current shell'
+export NVM_DIR="$HOME/.nvm"
+mkdir -p "$NVM_DIR"
+[ -s "$HOMEBREW_PREFIX/opt/nvm/nvm.sh" ] && \. "$HOMEBREW_PREFIX/opt/nvm/nvm.sh" --install
+
+echo '--- :node: 3. Install node version from .nvmrc'
+nvm install "$(cat .nvmrc)" && nvm use
+
+echo '--- :node: 4. nmp ci'
+npm ci
+
+echo '--- :ios: Set env var for iOS E2E testing'
+set -x
+export TEST_RN_PLATFORM=ios
+export TEST_ENV=sauce
+# Set a destination different from the hardcoded one which only works in the
+# older Xcode-setup used by CircleCI
+export RN_EDITOR_E2E_IOS_DESTINATION='platform=iOS Simulator,name=iPhone 14,OS=16.4'
+set +x
+
+echo '--- :react: Build iOS app for E2E testing'
+npm run core test:e2e:build-app:ios
+
+echo '--- :react: Build iOS bundle for E2E testing'
+npm run test:e2e:bundle:ios
+
+echo '--- :compression: Prepare artifact for SauceLabs upload'
+WORK_DIR=$(pwd) \
+  && pushd ./gutenberg/packages/react-native-editor/ios/build/GutenbergDemo/Build/Products/Release-iphonesimulator \
+  && zip -r "$WORK_DIR/gutenberg/packages/react-native-editor/ios/GutenbergDemo.app.zip" GutenbergDemo.app \
+  && popd
+
+echo '--- :saucelabs: Upload app artifact to SauceLabs'
+SAUCE_FILENAME=${BUILDKITE_BRANCH//[\/]/-}
+curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" \
+  --location \
+  --request POST 'https://api.us-west-1.saucelabs.com/v1/storage/upload' \
+  --form 'payload=@"./gutenberg/packages/react-native-editor/ios/GutenbergDemo.app.zip"' \
+  --form "name=Gutenberg-$SAUCE_FILENAME.app.zip" \
+  --form 'description="Gutenberg"'
```

Comment:
Just a reminder that we discussed moving off Sauce Labs and agreed it to be something to do in a standalone PR p9ugOq-41A-p2